url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/treemaker/XTreeMaker.java

```diff
@@ -0,0 +1,1486 @@
+				tokenizer.next();
+				l.add(makeImportEntry(from));
+			}
+		}
+		return l;
+	}
+
+	private XTreeImportEntry makeImportEntry(boolean from){
+		startPosition();
+		XTreeIdent as;
+		XTreeIdent ident = from?makeIdent():makeIdentEx();
+		if(tokenOk(XTokenKind.KEYWORD, XKeyword.AS)){
+			tokenizer.next();
+			as = makeIdent();
+		}else{
+			as = null;
+		}
+		return new XTreeImportEntry(endPosition(), ident, as);
+	}
+	
+	private XTreeExpr makeExpr(){
+		return makeExpr(false);
+	}
+	
+	private XTreeExpr makeExpr(boolean inList){
+		XTreeExpr expr = makeExprPrefixSuffix(inList);
+		XTreeExpr between = null;
+		while(isOperator()){
+			startPosition();
+			between = null;
+			XOperator o = readOperator(Type.INFIX, inList);
+			if(o==XOperator.IF){
+				between = makeExprPrefixSuffix(inList);
+				expect(XKeyword.COLON);
+			}
+			if(o==XOperator.NONE){
+				break;
+			}
+			XTreeExpr expr2  = makeExprPrefixSuffix(inList);
+			expr = mergeStatements(endPosition(), expr, o, expr2, between);
+		}
+		return expr;
+	}
+	
+	private XTreeExpr mergeStatements(XTreePosition position, XTreeExpr left, XOperator o, XTreeExpr right, XTreeExpr between){
+		if(left instanceof XTreeOperator){
+			XTreeOperator oLeft = (XTreeOperator) left;
+			if(oLeft.operators.get(0).priority<o.priority){
+				oLeft.exprs.set(oLeft.exprs.size()-1,  mergeStatements(position, oLeft.exprs.get(oLeft.exprs.size()-1), o, right, between));
+				return oLeft;
+			}else{
+				if(o==XOperator.IF)
+					return new XTreeOperatorIf(position, left, between, right);
+				if(oLeft.operators.get(0).priority==o.priority){
+					oLeft.operators.add(o);
+					oLeft.exprs.add(right);
+					return oLeft;
+				}else{
+					List<XTreeExpr> exprs = new ArrayList<XTreeExpr>();
+					exprs.add(left);
+					exprs.add(right);
+					List<XOperator> operators = new ArrayList<XOperator>();
+					operators.add(o);
+					return new XTreeOperator(position, exprs, operators);
+				}
+			}
+		}else if(left instanceof XTreeOperatorIf){
+			XTreeOperatorIf oLeft = (XTreeOperatorIf) left;
+			if(XOperator.IF.priority<o.priority){
+				oLeft.onFalse = mergeStatements(position, oLeft.onFalse, o, right, between);
+				return oLeft;
+			}else{
+				if(o==XOperator.IF)
+					return new XTreeOperatorIf(position, left, between, right);
+				List<XTreeExpr> exprs = new ArrayList<XTreeExpr>();
+				exprs.add(left);
+				exprs.add(right);
+				List<XOperator> operators = new ArrayList<XOperator>();
+				operators.add(o);
+				return new XTreeOperator(position, exprs, operators);
+			}
+		}else{
+			if(o==XOperator.IF)
+				return new XTreeOperatorIf(position, left, between, right);
+			List<XTreeExpr> exprs = new ArrayList<XTreeExpr>();
+			exprs.add(left);
+			exprs.add(right);
+			List<XOperator> operators = new ArrayList<XOperator>();
+			operators.add(o);
+			return new XTreeOperator(position, exprs, operators);
+		}
+	}
+	
+	private XTreeExpr makeExprPrefixSuffix(boolean inList){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.